src/app.jsx

```diff
@@ -357,3 +363,124 @@ export const ViewSelector = ({ isGrid, setIsGrid, sortBy, setSortBy }) => {
+                    <EllipsisVIcon />
+                </MenuToggle>}
+        >
+            <DropdownList>
+                <DropdownItem itemId='delete-item' key="delete-item" isDisabled={!selected} onClick={deleteItem} className={!selected ? "" : "pf-m-danger"}>
+                    {_("Delete")}
+                </DropdownItem>
+            </DropdownList>
+        </Dropdown>
+    );
+};
+
+const ConfirmDeletionDialog = ({ selected, itemPath }) => {
+    const Dialogs = useDialogs();
+
+    const deleteItem = () => {
+        if (selected.type === "file") {
+            cockpit.spawn(["rm", itemPath]).then(Dialogs.close)
+                    .catch(() => { Dialogs.show(<ForceDeleteModal itemPath={itemPath} />) });
+        } else if (selected.type === "directory") {
+            let emptyDirectory = false;
+            // Check for empty directory
+            cockpit.spawn(["ls", "-A", itemPath])
+                    .then(result => { emptyDirectory = result === "" })
+                    .then(() => {
+                        if (emptyDirectory)
+                            cockpit.spawn(["rmdir", itemPath]).then(Dialogs.close);
+                        else {
+                            cockpit.spawn(["rm", "-r", itemPath]);
+                        }
+                    })
+                    .then(Dialogs.close)
+                    .catch(() => { Dialogs.show(<ForceDeleteModal itemPath={itemPath} />) });
+        }
+    };
+
+    return (
+        <Modal
+            position="top"
+            title={_("Confirm deletion")}
+            isOpen
+            onClose={Dialogs.close}
+            footer={
+                <>
+                    <Button variant='danger' onClick={deleteItem}>{_("Delete Item")}</Button>
+                    <Button variant='link' onClick={Dialogs.close}>{_("Cancel")}</Button>
+                </>
+            }
+        >
+            {cockpit.format(_("Are you sure you want to permenantly delete $0?"), selected.name)}
+        </Modal>
+    );
+};
+
+const ForceDeleteModal = ({ itemPath }) => {
+    const Dialogs = useDialogs();
+
+    const forceDelete = () => {
+        cockpit.spawn(["rm", "-rf", itemPath])
+                .then(Dialogs.close)
+                .catch((err) => { Dialogs.show(<DeleteErrorModal errorMessage={err.message} />) });
+    };
+
+    return (
+        <Modal
+            position="top"
+            title={_("Force Delete")}
+            isOpen
+            onClose={Dialogs.close}
+            footer={
+                <>
+                    <Button variant='danger' onClick={forceDelete}>{_("Force Delete")}</Button>
+                    <Button variant='link' onClick={Dialogs.close}>{_("Cancel")}</Button>
+                </>
+            }
+        >
+            {_("Unable to delete item, do you want to force delete?")}
+        </Modal>
+    );
+};
+
+const DeleteErrorModal = ({ errorMessage }) => {
+    const Dialogs = useDialogs();
+
+    return (
+        <Modal
```

Comment:
Errors should appear inline in modals, use the component from pkg/lib/cockpit-components-inline-notification.jsx

See cockpit-machines/cockpit-podman for the usage.

Reply:
My point was not clear here ;) please don't create a new modal for showing the error. Just use the existing modal that attempted the deletion and show the inline errror notification in the start of the modal body.